tests/integration_tests/test_botorch.py

```diff
@@ -32,6 +33,9 @@
 
 @pytest.mark.parametrize("n_objectives", [1, 2, 4])
 def test_botorch_candidates_func_none(n_objectives: int) -> None:
+    if n_objectives == 1 and sys.version_info < (3, 8):
```

Comment:
The logic used to bypass the tests seems somewhat indirect. It could be clearer if it directly verifies the version of botorch, in a similar manner to what's demonstrated in `optuna/integration/botorch.py`. Would that be a feasible approach for you to consider?

https://github.com/optuna/optuna/blob/2b6addc365003eaab9e20bacb964ff15ed057da7/optuna/integration/botorch.py#L42